pkg/kubelet/dockertools/fake_docker_client.go

```diff
@@ -454,7 +457,14 @@ func (f *FakeDockerClient) InspectImageByRef(name string) (*dockertypes.ImageIns
 	defer f.Unlock()
 	f.appendCalled(calledDetail{name: "inspect_image"})
 	err := f.popError("inspect_image")
-	return f.Image, err
+	if result, ok := f.ImageInspects[name]; ok {
```

Comment:
nit: The logic here is a little bit anti-intuitive.

Will it break any existing test if we change this to:
```
if err != nil {
  return nil, err
}
result, ok := f.ImageInspects[name]
if !ok {
  return nil, ImageNotFoundError{name}
}
return result, nil
}
```